File: wrap.py
import sys

def main():
    if len(sys.argv) < 2:
        print('Wrap long lines in a wide terminal.');
        print('Usage: wrap <column>\nExample: ls -lah | wrap 60')
        sys.exit(0)

    col = int(sys.argv[1])

    try:
        for line in sys.stdin:
            while line:
                print(line[:col].strip())
                line = line[col:]
    except KeyboardInterrupt:
        pass

if __name__ == '__main__':
    main()

File: nim/wrap.nim
import os, strutils, sequtils

proc main(args: seq[string]) =
    if len(args) < 2:
        echo "Wrap long lines in a wide terminal."
        echo "Usage: wrap <column>\nExample: ls -lah | wrap 64"
        quit(0)
    
    let col = args[1].parseInt()

    for each_line in stdin.lines:
        var line = each_line
        while true:
            if col >= line.len():
                echo line
                break
            else:
                echo line[0 .. col - 1]
                line = line[col .. line.len() - 1]

when isMainModule:
    main concat(@[os.getAppFilename()], os.commandLineParams())
